src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ServiceDescriptionService.java

```diff
@@ -498,7 +528,178 @@ public ServiceDescriptionType refreshServiceDescription(Long id, boolean ignoreW
     }
 
     /**
-     * Get one ServiceDescriptionType by id
+     * Update Rest service description
+     *
+     * @param id
+     * @param url
+     * @param restServiceCode
+     * @param newRestServiceCode
+     * @return {@link ServiceDescriptionType}
+     * @throws UrlAlreadyExistsException
+     * @throws ServiceCodeAlreadyExistsException
+     */
+    public ServiceDescriptionType updateRestServiceDescription(Long id, String url, String restServiceCode,
+                                                               String newRestServiceCode)
+            throws UrlAlreadyExistsException, ServiceCodeAlreadyExistsException {
+        verifyAuthority("EDIT_REST");
+
+        if (newRestServiceCode == null) {
+            newRestServiceCode = restServiceCode;
+        }
+
+        ServiceDescriptionType serviceDescription = getServiceDescriptiontype(id);
+        serviceDescription.setUrl(url);
+        updateServiceCodes(restServiceCode, newRestServiceCode, serviceDescription);
+
+        checkDuplicateServiceCodes(serviceDescription);
+        checkDuplicateUrl(serviceDescription);
+
+        clientRepository.saveOrUpdateAndFlush(serviceDescription.getClient());
+        return serviceDescription;
+    }
+
+    /**
+     * Update OpenApi3 ServiceDescription
+     *
+     * @param id
+     * @param url
+     * @param restServiceCode
+     * @param newRestServiceCode
+     * @param ignoreWarnings
+     * @return
+     * @throws UrlAlreadyExistsException
+     * @throws ServiceCodeAlreadyExistsException
+     * @throws UnhandledWarningsException
+     * @throws OpenApiParser.ParsingException
+     */
+    public ServiceDescriptionType updateOpenApi3ServiceDescription(Long id, String url, String restServiceCode,
+            String newRestServiceCode, Boolean ignoreWarnings) throws UrlAlreadyExistsException,
+            ServiceCodeAlreadyExistsException, UnhandledWarningsException, OpenApiParser.ParsingException {
+
+        verifyAuthority("EDIT_OPENAPI3");
+        ServiceDescriptionType serviceDescription = getServiceDescriptiontype(id);
+
+        if (newRestServiceCode == null) {
+            newRestServiceCode = restServiceCode;
+        }
+
+        updateServiceCodes(restServiceCode, newRestServiceCode, serviceDescription);
+
+        // Parse openapi definition and handle updating endpoints and acls
+        if (serviceDescription.getUrl() != url) {
```

Comment:
Should use `!....equals()` instead of `!=`

Reply:
fixed